### PR TITLE
Custom evosuite build process 

### LINKED ISSUE #33: Create separate build process for custom evosuite builds
# Description
Since this plugin relies on a custom build of evosuite to work, the jar is currently sources as a raw binary inside the project's root which is **not** a good practice. Create a separate build process that allows developers to make custom build releases and integrate it with TestGenie's own build process.
# Definition of Done
* [x]  Custom Evosuite build has a separate CI process with artifact releases.
* [x]  TestGenie's build scripts are extended to leverage the above and allows the developers to seamlessly update their evosuite binary without littering the git history.

### PR DESCRIPTION
**Pull request by lyuben-todorov**
# Description of changes made
* Added a github workflow for building custom versions for evosuite here https://github.com/ciselab/evosuite/blob/thunderdome/.github/workflows/release.yml. It is triggered whenever a developer pushes to a branch with a name matching the pattern `release/<version>`, producing a release tagged with <version>. The build artifacts/jar can be downloaded with the github api.
* Added a custom gradle task to the TestGenie project which executes before any compilation tasks. The task is supplied with a version number which is defined at the top of `build.gradle.kts`. Afterewards it attempts to fetch the release corresponding to that version from https://github.com/ciselab/evosuite/releases unzips it and provides the evosuite jar to the build and packaging process.

# Why is merge request needed
Such a feature would be very useful since storing large binaries such as jar files in the project sources is _not_ a good idea. This PR not only solves that problem but also simplifies working with custom builds of evosuite.
# Other notes
Project documentation will be updated eventually to explain in detail how the process of updating and releasing a new build of evosuite works.

closes #33

### CODE REVIEW

File: src/main/kotlin/com/github/mitchellolsthoorn/testgenie/evosuite/Runner.kt [hunk 1]
```diff
@@ -20,7 +20,7 @@ class Runner {
             val javaPath = "java"// TODO: Source from config
 
             val pluginsPath = System.getProperty("idea.plugins.path")
-            val evoSuitePath = "$pluginsPath/TestGenie/lib/evosuite.jar"
+            val evoSuitePath = "$pluginsPath/TestGenie/lib/evosuite-1.0.2.jar"
```
Comment (jzelenjak) [inline src/main/kotlin/com/github/mitchellolsthoorn/testgenie/evosuite/Runner.kt:23]: I think you had a variable for this version. Wouldn't it make sense to use it here?

File: build.gradle.kts [hunk 2]
```diff
@@ -1,8 +1,13 @@
 import org.jetbrains.changelog.markdownToHTML
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import java.net.URL
+import java.util.zip.ZipInputStream
+import java.io.FileOutputStream
 
 fun properties(key: String) = project.findProperty(key).toString()
 
+val thunderdomeVersion = "1.0.2"
```
Comment (jzelenjak) [inline build.gradle.kts:9]: Is there any specific meaning for this variable name?

Comment (jzelenjak) [reply build.gradle.kts:9]: Maybe it can be changed to something more descriptive?

**Review by jzelenjak — COMMENTED** [review]
Looks very good. The changes you have made are very useful for our project. I don't have any comments here other than the 2 i have left. Good job!

**Review by SergeyDatskiv — COMMENTED** [review]
To me it seems like the changes here are a part of the changes from pull request #35. Everything looks alright. I will approve the request once Jegor's comments are resolved. 

Thank you for your work. Good job! 🙂 👍

File: src/main/kotlin/com/github/mitchellolsthoorn/testgenie/evosuite/Runner.kt (hunk 1 above)
Comment (kirilvasilev16) [reply src/main/kotlin/com/github/mitchellolsthoorn/testgenie/evosuite/Runner.kt:23]: I also agree with this ☝

**Review by kirilvasilev16 — APPROVED** [review]

**Review by jzelenjak — APPROVED** [review]

**Pull request merged**